In [12]:
def checkPalindrome(num):
#Implement Your Code Here
	pass
		
n=int(input())
temp=n
rev=0
while(n>0):
    dig=n%10
    rev=rev*10+dig
    n=n//10
if(temp==rev):
    print("true")
else:
    print("false")


232
true


In [13]:
def checkPalindrome(num):
#Implement Your Code Here
	pass
		
n=int(input())
temp=n
rev=0
while(n>0):
    dig=n%10
    rev=rev*10+dig
    n=n//10
if(temp==rev):
    print("true")
else:
    print("false")



244456
false


In [14]:
def checkPalindrome(num):
#Implement Your Code Here
	pass
		
n=int(input())
temp=n
rev=0
while(n>0):
    dig=n%10
    rev=rev*10+dig
    n=n//10
if(temp==rev):
    print("true")
else:
    print("false")



11001100110011
true


In [28]:
n = int(input())
j=1
while j<=n:
    i=1
    while i<=n:
        print("*",end='')
        i=i+1
    print()
    j=j+1
    


2
**
**


In [32]:
n = int(input())
j=1
while j<=n:
    i=1
    while i<=n:
        print("*",end='')
        i=n-i
    print()
    j=j+1

4
**************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************

KeyboardInterrupt: 

In [33]:

n = int(input())
j=1
while j<=n:
    i=1
    while i<=n:
        print("7",end='')
        i=i+1
    print()
    j=j+1

7
7777777
7777777
7777777
7777777
7777777
7777777
7777777


In [35]:

n = int(input())
j=1
while j<=n:
    i=1
    while i<=n:
        print(int(n),end='')
        i=i+1
    print()
    j=j+1
    

4
4444
4444
4444
4444


In [55]:

n = int(input())
j=1
while j<=n:
    i=1
    while i<=n:
        print(i, end=" ")
        i=i+1
    print()
    j=j+1

4
1 2 3 4 
1 2 3 4 
1 2 3 4 
1 2 3 4 


In [113]:
n = int(input())
j=1
while j<=n:
    i=1
    while i<=n:
        print("*", end=" ")
        i=i+1
    print(" ")
    j=j+1

5
* * * * *  
* * * * *  
* * * * *  
* * * * *  
* * * * *  


In [151]:
n = int(input())

i=1
while i<=n:
    j=1
    while j<=i:
        print(j,end=" ")
        j=j+1
    print()
    i=i+1

5
1 
1 2 
1 2 3 
1 2 3 4 
1 2 3 4 5 


In [155]:
n = int(input())

i=1
while i<=n:
    j=1
    while j<=n-i+1:
        print(j,end=" ")
        j=j+1
    print()
    i=i+1

5
1 2 3 4 5 
1 2 3 4 
1 2 3 
1 2 
1 


1
23
345
4567

In [175]:
n = int(input())

i=1

while i<=n:
    j=1
    p=i
    while j<=i:
        print(p,end=" ")
        p=p+1
        j=j+1
    print()
    i = i+1

4
1 
2 3 
3 4 5 
4 5 6 7 


1
23
456
78910

In [180]:
n = int(input())

i=1
p=1
while i<=n:
    j=1
    while j<=i:
        print(p,end=" ")
        p=p+1
        j=j+1
    print()
    i = i+1

4
1 
2 3 
4 5 6 
7 8 9 10 


Print the following pattern for the given N number of rows.

Pattern for N = 4
*
**
***
****


Input format  : Integer N (Total no. of rows)
Output format : Pattern in N lines

In [182]:
n = int(input())

i = 1

while i<=n:
    j=1
    while j<=i:
        print("*",end="")
        j=j+1
    print()
    i=i+1

4
*
**
***
****


Print the following pattern for the given N number of rows.

Pattern for N = 4
1
22
333
4444

Input format : Integer N (Total no. of rows)
Output format: Pattern in N lines


In [183]:
n = int(input())
i = 1
while i<=n:
    j=1
    while j<=i:
        print(i,end="")
        j=j+1
    print()
    i=i+1

4
1
22
333
4444


Print the following pattern for the given N number of rows.

Pattern for N = 4
1
21
321
4321

Input format : Integer N (Total no. of rows)
Output format: Pattern in N lines

In [188]:
n = int(input())
i = 1
while i<=n:
    j=1
    p=i
    while j<=i:
        print(p,end="")
        p=p-1
        j=j+1
    print()
    i=i+1

4
1
21
321
4321


Print the following pattern for the given N number of rows.

Pattern for N = 3
 A
 BB
 CCC
 
Input format : Integer N (Total no. of rows)
Output format: Pattern in N lines


In [194]:



n = int(input())
i = 1
while i<=n:
    j=1
    k=i
    charP = chr(ord('A')+k-1)
    while j<=i:
        print(charP,end="")
        k=k+1
        j=j+1
    print()
    i=i+1

4
A
BB
CCC
DDDD


In [ ]:
Print the following pattern for the given N number of rows.

Pattern for N = 4

A
BC
CDE
DEFG

Input format : Integer N (Total no. of rows)
Output format: Pattern in N lines


In [204]:
n = int(input())
i = 1
while i<=n:
    j=1
    k=i
    while j<=i:
        charP = chr(ord('A')+k-1)
        print(charP,end="")
        k=k+1
        j=j+1
    print()
    i=i+1

4
A
BC
CDE
DEFG


Print the following pattern for the given number of rows.

Pattern for N = 5

E
DE
CDE
BCDE
ABCDE

Input format : N (Total no. of rows)
Output format : Pattern in N lines

In [216]:
n = int(input())
i = 1
while i<=n:
    j=1
    k=n
    while j<=i:
        charP = chr(ord('A')+k-i)
        print(charP,end="")
        k=k+1
        j=j+1
    print()
    i=i+1

5
E
DE
CDE
BCDE
ABCDE


Print the following pattern for the given N number of rows.

Pattern for N = 4

4444
333
22
1

Input format : Integer N (Total no. of rows)
Output format:Pattern in N lines

In [238]:
n = int(input())
i = 1
while i<=n:
    j=1
    while j<=n-i+1:
        p=n-i+1
        print(p,end="")
        j=j+1
    print()
    i=i+1
    
    
    i,space (3,)

4
4444
333
22
1


In [291]:
n = int(input())
i=1
while i<=n:
    space=1
    while space<=n-i:
        print(" ",end="")
        space=space+1
    stars=1 
    while stars<=i:
            print("*",end="")
            stars=stars+1
    print()
    i=i+1
    

7
      *
     **
    ***
   ****
  *****
 ******
*******


In [292]:
n=int(input())
i=1
while i<=n:
    space=1
    while space<=n-i:
        print(" ",end="")
        space=space+1
    count=1
    while count<=i:
        print(count,end="")
        count=count+1
    print()
    i=i+1
    

7
      1
     12
    123
   1234
  12345
 123456
1234567


In [321]:
n=int(input())
i=1

while i<=n:
    space=1
    while space<=n-i:
        print(" ",end="")
        space=space+1
    p=1
    j=1
    while p<=i:
        print(p,end="")
        p=p+1
    p=i-1
    while p>=1:
        print(p,end="")
        p=p-1
    print()
    i=i+1


6
     1
    121
   12321
  1234321
 123454321
12345654321


In [322]:
n=int(input())
i=1

while i<=n:
    space=1
    while space<=n-i:
        print(" ",end="")
        space=space+1
    p=1
    j=1
    while p<=i:
        print("*",end="")
        p=p+1
    p=i-1
    while p>=1:
        print("*",end="")
        p=p-1
    print()
    i=i+1



5
    *
   ***
  *****
 *******
*********


  1
 232
34543

In [385]:
n=int(input())
i=1

while i<=n:
    space=1
    while space<=n-i:
        print(" ",end="")
        space=space+1
    p=n-i
    while p<=n and p>=1:
        print(p,end="")
        p=p-1
    p=n
    while p>=1 and p<=n:
        print(p,end="")
        p=p-1

    print()
    i=i+1

5
    432154321
   32154321
  2154321
 154321
54321


In [2]:
n=int(input())
i=1

while i<=n:
    space=1
    while space<=n-i:
        print(" ",end="")
        space=space+1
    p=n-i
    while p>=1:
        print(p,end="")
        p=p-1
    print()
    i=i+1

5
    4321
   321
  21
 1



In [4]:
n=int(input())

i=1
while i<=n:
    space=1
    while space<=n-i:
        print(" ",end="")
        space=space+1
    p1=1
    while p1<=n:
        print("*",end="")
        p1=p1+1
    p2=i-1
    while p2>=1:
        print("*",end="")
        p2=p2-1 
    print()
    i=i+1

5
    *****
   ******
  *******
 ********
*********


In [4]:
n = int(input())

for i in range(1,n+1,1):
    print(i)

5
1
2
3
4
5


by default start is at 0, default step is 1

range(start, stop, stride)

stride can be negative for decrement
stop will be excluded
start is included always

In [9]:
a = int(input())
b = int(input())

for i in range(a,b+1,1):
    if i%3==0:
        print(i, end=", ")

1
100
3, 6, 9, 12, 15, 18, 21, 24, 27, 30, 33, 36, 39, 42, 45, 48, 51, 54, 57, 60, 63, 66, 69, 72, 75, 78, 81, 84, 87, 90, 93, 96, 99, 

In [17]:
n=int(input())
flag=False
for d in range(2,n,1):
    if n%d==0:
        flag=True
if flag:
    print("not prime")
else:
    print("prime")

89438438
not prime


In [39]:
n = int(input())

u=1
for i in range(1,n+1,1):
    for s in range(n-i):
        print(" ",end="")
    for j in range(i,2*i,1):
        print(j,end="")
    for j in range(2*i-2,i-1,-1):
        print(j,end="")
    print()
    u=u+1

for i in range(1,n+1,1):
      for u in range(2*i-2,i-1,-1):
        print(u,end="")
        
        

5
    1
   232
  34543
 4567654
567898765
2436548765

In [ ]:
n = int(input())

while i<n:
    for i in range(2+i*3,1001):
        if i%4!=0:
            print(i,end=" ")

In [ ]:
n = int(input())

while n%2==0:

In [6]:
n = int(input())
sum2=[]
for i in range(1,1001,1):
    sum1=0
    sum1=3*i+2
    if sum1%4!=0:
        sum2.append(sum1)
    if len(sum2)==n:
        break
for i in sum2:
    print(i,'',end='')

10
5 11 14 17 23 26 29 35 38 41 

In [9]:
s = input()
a = 'x'
b = 'y'

p = s.find(a) 
ans = s

if p != -1:
    ans = s[:p]+b+s[p+1:]
    print(ans)

swedxoi
swedyoi


In [12]:
s = input()
a = 'x'
b = 'y'

ans = s

for i in range(0, len(s)):
    if s [i] == a:
        ans = s[:i]+b+s[i + 1:]
        print(ans)

fjfkfxkdk
fjfkfykdk


In [39]:
s = "abcdefghi"
s.split

TypeError: 'builtin_function_or_method' object is not subscriptable

In [45]:
S = input()

so = S[::-1] 
if S == so:
        print("True")
else:
        print("False")

dsdd
False


In [67]:
s = input()

n = len(s)
i = 0
j = i+1
k = i+2
while i<=n:
    print(s[i])
    print(s[i]+s[j])
    print(s[i]+s[j]+s[k])
    i = i + 1
    j = j + 1
    k = k + 1
    break
    


xyz
x
xy
xyz


In [4]:
s = input()

def reverse(s): 
    return s[::-1] 
  
def isPalindrome(s):  
    rev = reverse(s) 
  
    if (s == rev): 
        return True
    return False
  

ans = isPalindrome(s) 
  
if ans == 1: 
    print("Yes") 
else: 
    print("No") 

sds
Yes


In [126]:
ne = input()
n = ne.split()

for i in range(0,len(n)-1,2):
    n[i],n[i+1]=n[i+1],n[i]
    i = i + 1
n

for i in range(len(n)):
    print(n[i],'',end='')

1 2 3 4
2 1 4 3 

In [4]:
n = input()
ne = n.split()
i = 2
ne.split(x,[i,len(n)])
ne.split(y,[0,i])
x
y


1 2 3 5


AttributeError: 'list' object has no attribute 'split'

In [32]:
ne = input().split()

ne = [int(i) for i in ne]

count = 0
for i in range(0,len(ne)+1):
    if sum(ne[0:i])==sum(ne[i:len(ne)+1]):
        count = count + 1
print(count)

-7 1 5 2 -4 3 0
3


#### 

In [49]:
ne = input().split()

ne = [int(i) for i in ne]

if sum(ne[0:i])-sum(ne[i+1:len(ne)+1])==0:
    print(i)
else:
    print('-1')
    
    

-7 1 5 2 -4 3 0
-1


In [96]:
m,n = input().split()

arr = []
arr = input().split()
arr = [int(i) for i in arr]



for i in range(0,len(arr),int(n)):
    result = sum(arr[i:int(n)+i])
    print(result, end=" ")
    

4 2
1 2 3 4 5
3 7 5 

range(0,len(arr)+1,int(n))

In [118]:
arr = []
arr = input().split()

m = arr.pop(0)
n = arr.pop(0)

matrix = [i for range(m)]
arr = [int(i) for i in arr]


for i in range(0,len(arr),int(n)):
    print(arr[i], end = " ")
    

3 4 1 2 3 4 5 6 7 8 9 10 11 12
1 5 9 

In [130]:
m = 3
n = 4
matrix = [[0 for i in range(m)] for j in range(n)]

In [128]:
matrix = arr

In [129]:
matrix

[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]

In [148]:
string = input()

j = 0
for i in range(len(string)-1):
    if string[i]==string[j]:
        string[i]=='/n'
        j = j + 1
string
    

refree


'refree'

Welcome to Coding Ninjas


In [269]:
string = input().split()

n = 0



Welcome to Patiyala
W o l o 

IndexError: string index out of range

In [196]:
string = input()
j = 0
for j in range(len(string)):
    print(string[-j-1], end="")
    

Welcome to Coding Ninjas
sajniN gnidoC ot emocleW

In [241]:
n = int(input())

for i in range(1,n+1):
    if i%2!=0:
        print(i,end=' ')
for i in range(1,n+1):
    if i%2==0:
        print(i,end=' ')

6
1 3 5 2 4 6 

In [225]:
range(1,5,-1)

range(1, 5, -1)

In [1]:
b = input()
a = input().strip()
b = len(a)


for i in range(len(a)-1):
    if a[i]>a[i+1]:
        print(a[i], end=" ")
    else
    continue
        

SyntaxError: invalid syntax (<ipython-input-1-5ed366161ab4>, line 9)

In [5]:
def fact(n):
    n_fact = 1
    for i in range(1,n+1):
        n_fact = n_fact * i
    return n_fact

a = int(input())
print(fact(a))

5
120


In [16]:
def printTable(start,end,step):
#Fahrenheit to celcius converter function

    for i in range(start, end+1, step):
        print(i, int((i-32)*5/9))
        pass 

s = int(input())
e = int(input())
step = int(input())
printTable(s,e,step)








0
100
20
0 -17
20 -6
40 4
60 15
80 26
100 37


In [ ]:
def checkMember(n):
#Implement Your Code Here
arr = range()
    for i in range(0,n+1):
        if 
    pass

n=int(input())
if(checkMember(n)):
    print("true")
else:
    print("false")

In [3]:
import pandas as pd

In [4]:
zomo = pd.read_csv("zomato.csv")

In [6]:
a = zomo["approx_cost(for two people)"]

In [7]:
a.head()

0    800
1    800
2    800
3    300
4    600
Name: approx_cost(for two people), dtype: object

In [11]:
a1 = a.str.replace(",","")

In [12]:
a2 = pd.to_numeric(a1)

In [13]:
a2

0         800.0
1         800.0
2         800.0
3         300.0
4         600.0
          ...  
51712    1500.0
51713     600.0
51714    2000.0
51715    2500.0
51716    1500.0
Name: approx_cost(for two people), Length: 51717, dtype: float64

In [17]:
ma = a2.max()
avg = a2.mean()
mi = a2.min()

In [21]:
print("max value is",ma)
print("min value is",mi)
print("avg value is",avg)

max value is 6000.0
min value is 40.0
avg value is 555.4315664479959


In [27]:
sample_size = 100
sample = a2.sample(sample_size, random_state = 1)

In [28]:
sample.shape

(100,)

In [32]:
, , 

(100.0, 3000.0, 565.0)

In [33]:
print("max value is",ma, sample.max())
print("min value is",mi, sample.min())
print("avg value is",avg, sample.mean())

max value is 6000.0 3000.0
min value is 40.0 100.0
avg value is 555.4315664479959 565.0


In [34]:
fund = pd.read_csv("0000000000002370.csv")

In [36]:
sample = 50

In [39]:
fund["CityLocation"]= fund["CityLocation"].str.replace("New Delhi", "Delhi")

In [50]:
fund["CityLocation"] = fund["CityLocation"].str.replace("bangalore", "Bangalore")